src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -546,13 +551,12 @@ public int MaxResponseDrainSize
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing && !_disposed)
+            if (!_disposed)
```

Comment:
The current WinHttpHandler uses SYNC pattern for WinHTTP.  But I agree when #3000 is done then we will have to consider this issue.